### PR TITLE
Viglink

### PR DESCRIPTION
**Pull request by michaelcullum**
Needs more tests but wanted to get the PR up.

Squash on merge.

### CODE REVIEW

File: src/AppBundle/Controller/VigLinkController.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace AppBundle\Controller;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\{Response, Request};
+
+class VigLinkController extends Controller
+{
+	public function keyAction(Request $request): Response
```
Comment (paul999) [inline src/AppBundle/Controller/VigLinkController.php]: We have PHP7 on .com?

Comment (michaelcullum) [inline src/AppBundle/Controller/VigLinkController.php]: Nope, and that's why we have travis (to slap me when I do stuff like that)

File: src/AppBundle/Controller/VigLinkController.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	{
+		$sitename = $request->query->get('sitename');
+		$siteId = $request->query->get('uuid');
+		$apiKey = $request->query->get('key');
+
+		if (!isset($sitename, $siteId, $apiKey))
+		{
+			throw new Symfony\Component\HttpKernel\Exception\NotFoundHttpException();
```
Comment (michaelcullum) [inline src/AppBundle/Controller/VigLinkController.php]: This is ugly but will fix when I add tests

**Comment by michaelcullum** [comment]
Fixed and tests pass

@marc1706

File: app/config/parameters_travis.yml [hunk 3]
```diff
@@ -66,5 +66,7 @@ parameters:
     # Location should be local, sandbox or live
     location:           local
 
+    viglink_secret_key: abc12345
```
Comment (marc1706) [inline app/config/parameters_travis.yml]: Mind referring me to the place where we actually have to define the correct one?

Comment (Derky) [inline app/config/parameters_travis.yml]: @marc1706 the correct one should be defined in `app/config/parameters.yml` 
(local file, not part of the repo) 

The same config entry should be added to `app/config/parameters.dist.yml` as well, but with a placeholder value like: `~`

File: src/AppBundle/Controller/VigLinkController.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		$key = $this->container->hasParameter('viglink_api_key') ? $this->getParameter('viglink_api_key') : $apiKey;
+
+		if (!$this->container->hasParameter('viglink_secret_key')) {
+			throw new Exception('Viglink is not enabled');
+		}
+
+		$secretKey = $this->getParameter('viglink_secret_key');
+
+		$url = ("https://www.viglink.com/users/convertAccount?key=" . $key . "&subId=" . $subId);
+		$signature = bin2hex(hash_hmac('md5', ($url . "-" . $expiration), $secretKey));
```
Comment (marc1706) [inline src/AppBundle/Controller/VigLinkController.php]: The hex info seems to be an error on viglink's side. See the example in the PDF and this:
https://3v4l.org/X6KHb

Therefore we should just do hash_hmac() and no bin2hex() around it.

Comment (unknownbliss) [inline src/AppBundle/Controller/VigLinkController.php]: Indeed. I couldn't quite understand the reasoning behind the hex stuff or what it was supposed to mean. I can push this fix evening EST, add proper tests and merge this.

**Pull request merged**